### PR TITLE
Upgrade cleanup for PrepareSteps

### PR DESCRIPTION
**Pull request by JingJZ160**
### Description
Further clean up of the foreshadow code, mostly focusing on the `PrepareSteps` as there are code we can extract to the parent class.

I did not extract all possible duplicate code, mostly because of readability concerns. If we do the full extraction for the `.fit` method, it will look exactly like the old foreshadow where we need to jump between multiple files to understand code. 

Let me know what you think.

### CODE REVIEW

**Comment by JingJZ160** [comment]
Ping :)

File: foreshadow/steps/flattener.py [hunk 1]
```diff
@@ -75,12 +50,16 @@ def transform(self, X, *args, **kwargs):
             A transformed dataframe.
 
         """
-        # if self._empty_columns is None:
-        #     raise ValueError("Cleaner has not been fitted yet.")
+        return super().transform(X=X)
```
Comment (jichaogp) [inline foreshadow/steps/flattener.py]: Do we still need to override this method?

File: foreshadow/steps/mapper.py [hunk 2]
```diff
@@ -66,28 +52,8 @@ def transform(self, X, *args, **kwargs):
...
-            X: the data frame.
-            *args: positional args.
-            **kwargs: key word args.
-
-        Returns:
-            A transformed dataframe.
-
         """
-        return self.fit(X, *args, **kwargs).transform(X)
+        return super().transform(X=X)
```
Comment (jichaogp) [inline foreshadow/steps/mapper.py]: Do we need to override this method then?

File: foreshadow/steps/flattener.py (hunk 1 above)
Comment (JingJZ160) [reply foreshadow/steps/flattener.py]: No I'll remove it.

File: foreshadow/steps/mapper.py (hunk 2 above)
Comment (JingJZ160) [reply foreshadow/steps/mapper.py]: Same as above.

**Pull request merged**